Antennas iq

Doreban opened the pull request:
Added antennas iq plotting to the data flow.

On the site computers:
- Added an if statement in the `site-linux.daemon` to run the borealis-data-utils `plot_antennas_iq ` function if it exists to create iq plots.
- Added to `rsync_to_campus` a check to find generated iq plots and transfer to the dataman user on campus. 

On the campus computers:
- created a `archive_iq_plots` to compress files that are over a day old into thumbnail images. Reducing the data size of each image from ~50 KB to ~ 2 KB.
- Amended `campus.daemon` to run `archive_iq_plots`.

Currently the addition of iq plots into the data flow is dependant on the borealis-data-utils repo. If reviewers think that the iq plotting should be rewritten into the data flow repo let me know.

tjk584 commented on campus/archive_iq_plots:
```diff
@@ -0,0 +1,84 @@
+# the other dataflow services in the campus daemon.
+#
+# Dependencies:
+#   - convert linux utility
```
Is this installed via zypper?

Doreban replied:
I believe it is default on at least opensuse operating systems. All computers I've checked on have had convert

tjk584 commented on campus/archive_iq_plots:
```diff
@@ -0,0 +1,84 @@
+readonly LOGGING_DIR="${HOME}/logs/archive_iq/$(date +%Y/%m)"
+mkdir --parents $LOGGING_DIR
+readonly LOGFILE="${LOGGING_DIR}/${RADAR_ID}.$(date +%Y%m%d).archive_iq.log"
+readonly  SUMMARY_DIR="${HOME}/logs/archive_iq/summary/$(date +%Y/%m)"
+mkdir --parents $SUMMARY_DIR
+readonly SUMMARY_FILE="${SUMMARY_DIR}/${RADAR_ID}.$(date -u +%Y%m%d).archive_iq.log"
```
Summary file was initially created for logs that are sent to logman for getting stats for the dashboard. We may not need these for this script.

tjk584 commented on inotify_daemons/site-linux.daemon:
```diff
@@ -46,6 +46,10 @@ do
 
         printf "Executing convert_and_restructure...\n"
         $HOME/data_flow/site-linux/convert_and_restructure
+        if [[ -f "${HOME}/borealis-data-utils/plot_antennas_iq" ]]; then
+            printf "Generating antennas iq plots...\n"
+            $HOME/borealis-data-utils/plot_antennas_iq
+        fi
```
I think we should add this script to the data_flow repo, since this is the only place it will be run in our regular operations

tjk584 reviewed (commented):
Looks pretty good so far, one suggestion:
- I think the antennas iq plotting script should be moved to this repo. Logging for this script should be made similar to the other dataflow scripts too, so we can easily know when/why plots don't make it to the dashboard.

Doreban commented:
Moved the iq plotting that was done in borealis-data-utils into the site-linux section of the data flow. I am not certain of how much of the new plotting scripts I added are absolutely required for the plotting, but I pulled the functions that were used and stitched it together and made sure it works the same as it did in the borealis-data-utils branch.

Doreban commented on site-linux/rsync_to_campus:
```diff
@@ -146,22 +145,28 @@ done
+  # Check if transfer was okay using the md5sum program
+  verify_transfer $file "${PLOTDEST}/${image}" $SDCOPY
+  return_value=$?
+  if [[ $eturn_value -eq 0 ]]; then
```
Yeah, but that's because it just interpreted the misspelled variable as empty so it passed the condition.

Doreban commented on site-linux/plot_antennas_iq:
```diff
@@ -0,0 +1,150 @@
+# Define directories
+readonly DATA_DIR=/borealis_nfs/borealis_data/antennas_iq_array/
+readonly FAILED_FILE_DEST=/borealis_nfs/borealis_data/conversion_failure/
+readonly PLOT_DEST=${HOME}/logging/daily_plots/
+readonly FULLSIZE_DEST=${HOME}/logging/daily_plots/fullsize/
+readonly FAILED_RESIZE_DEST=${HOME}/logging/daily_plots/failed_resize/
```
I'll move it over then. I just wasn't sure if they should be still kept under logging or not.

tjk584 commented on README.md:
```diff
@@ -26,15 +26,21 @@ file starts being written
+3. site-linux/plot_antennas_iq: reads in the antennas iq files after the previous script has run.
+Creates plots of the iq data for each rx path using the last generated file. Triggered via 
+site-linux.daemon when convert_and_restructure finishes executing.
+4. site-linux/rsync_to_campus: Moves rawacf DMAP, array files, and i1 plots from the site storage to 
+the university campus server at sdc-serv. Triggered via site-linux.daemon when plot_antennas_iq
+finishes executing. 
```
```suggestion
4. site-linux/rsync_to_campus: Moves rawacf DMAP, array files, and antennas iq plots from the site storage to 
the university campus server at sdc-serv. Triggered via site-linux.daemon when plot_antennas_iq
finishes executing. 
```

tjk584 commented on site-linux/plot_antennas_iq:
```diff
@@ -26,9 +26,9 @@ source "${HOME}/pydarnio-env/bin/activate"
-readonly PLOT_DEST=${HOME}/logging/daily_plots/
-readonly FULLSIZE_DEST=${HOME}/logging/daily_plots/fullsize/
-readonly FAILED_RESIZE_DEST=${HOME}/logging/daily_plots/failed_resize/
+readonly PLOT_DEST=${HOME}/logs/daily_plots/
+readonly FULLSIZE_DEST=${HOME}/logs/daily_plots/fullsize/
+readonly FAILED_RESIZE_DEST=${HOME}/logs/daily_plots/failed_resize/
```
```suggestion
readonly PLOT_DEST=${HOME}/logs/daily_plots/
readonly FULLSIZE_DEST=${PLOT_DEST}/fullsize/
readonly FAILED_RESIZE_DEST=${PLOT_DEST}/failed_resize/
```

tjk584 reviewed (changes requested):
Looks good, just one typo to fix
